### PR TITLE
Lesson 4 1

### PR DESCRIPTION
**Pull request by BThomas22tech**

### CODE REVIEW

File: src/components/AddTodoForm.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import React from "react";
 import InputWithLabel from "./InputWithLabel";
+import PropTypes from "prop-types"
 
 function AddTodoForm(props) {
+  AddTodoForm.propTypes = {
```
Comment (riousenkai) [inline src/components/AddTodoForm.js]: We shouldn't declare propTypes inside the component (please update the other files as well).

File: src/components/AddTodoForm.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import React from "react";
 import InputWithLabel from "./InputWithLabel";
+import PropTypes from "prop-types";
 
+AddTodoForm.propTypes = {
+  onAddTodo: PropTypes.func.isRequired,
+};
 function AddTodoForm(props) {
```
Comment (riousenkai) [inline src/components/AddTodoForm.js:9]: You should have a space between proptypes and the function, but you can make that change in your final project.

File: src/components/TodoList.js [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
+import PropTypes from "prop-types"
 
+TodoList.propTypes = {
+  title: PropTypes.string.isRequired,
+  key: PropTypes.string.isRequired,
+}
 function TodoList(props) {
   return (
-    <ul>
+    <ul >
```
Comment (riousenkai) [inline src/components/TodoList.js]: Remove the unnecessary space

**Comment by riousenkai** [comment]
I'm getting these errors in my console - 
image

**Comment by BThomas22tech** [comment]
> I'm getting these errors in my console - image

Can I take out the isRequired? Or is that needed in this case?

**Comment by riousenkai** [comment]
@BThomas22tech - even if you remove it, it's still wrong. 

I would advice that you destructure your props (this is a requirement for your final anyway) - that should help you see what's wrong

**Comment by BThomas22tech** [comment]
> @BThomas22tech - even if you remove it, it's still wrong.
> 
> I would advice that you destructure your props (this is a requirement for your final anyway) - that should help you see what's wrong

is this the right direction? `TodoList.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

function TodoList({todoList,onRemoveTodo}) {
  return (
    <ul>
      {todoList.map(({ title,id }) => {
        return (
          <TodoListItem
            key={id}
            title={title}
            onRemoveTodo={() => onRemoveTodo(id)}
          />
        );
      })}
    </ul>
  );
}`

a little confused beyond that.

**Comment by riousenkai** [comment]
@BThomas22tech  - Take a look at your destructured props and the prop types you declared.

**Comment by BThomas22tech** [comment]
> a88da88

is the updated code

**Review by riousenkai — APPROVED** [review]

**Pull request merged**